library/std/src/sys/unix/thread.rs

```diff
@@ -84,6 +96,13 @@ impl Thread {
             };
         }
 
+        #[cfg(target_os = "horizon")]
+        {
+            // Set the priority and affinity values
+            assert_eq!(libc::pthread_attr_setpriority(&mut attr, priority), 0);
```

Comment:
Would it make sense to use this [existing API](https://man7.org/linux/man-pages/man3/pthread_attr_setschedparam.3.html) which is defined on some platforms?

Reply:
I can try to route things through those interfaces.

Reply:
If this fails we could also make our pthread extensions more noticably non-portable by adding the `_np` suffix like those affinity functions do.

Reply:
There's some interesting functions here: https://man7.org/linux/man-pages/man7/sched.7.html

Reply:
This has been updated to use a standard API for priority, but the affinity/"ideal processor" API isn't standard anywhere else, so I've made an explicitly non-portable (np) version.